frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -435,15 +529,42 @@ class SchemaFinder extends Visitor {
       primaryColumnName: 'id',
       foreignTableName: foreignTableName,
     })
+    const foreignKeyConstraint = aForeignKeyConstraint({
+      targetTableName: primaryTableName,
+      targetColumnName: 'id',
+    })
 
-    extractForeignKeyOptions(argNodes, relationship)
+    extractForeignKeyOptions(argNodes, relationship, foreignKeyConstraint)
 
     this.relationships.push(relationship)
+    const foreignTable = this.tables.find(
+      (table) => table.name === foreignTableName,
+    )
+    if (foreignTable) {
+      foreignTable.constraints[foreignKeyConstraint.name] = foreignKeyConstraint
+    }
+  }
+
+  handleAddCheckConstraint(node: CallNode): void {
+    const argNodes = node.arguments_?.compactChildNodes() || []
+
+    const constraintResult = extractCheckConstraint(argNodes)
+    if (constraintResult.isErr()) {
+      this.errors.push(constraintResult.error)
+      return
+    }
+    const { tableName, constraint } = constraintResult.value
+    const table = this.tables.find((table) => table.name === tableName)
+    if (table) {
+      table.constraints[constraint.name] = constraint
+    }
   }
 
   override visitCallNode(node: CallNode): void {
     if (node.name === 'create_table') this.handleCreateTable(node)
     if (node.name === 'add_foreign_key') this.handleAddForeignKey(node)
+    if (node.name === 'add_check_constraint')
+      this.handleAddCheckConstraint(node)
```

Comment:
In recent versions of Rails (7.1 and later) and with the mysql2 adapter, t.check_constraint is now supported and will be included in schema.rb.👀
like this:
```
create_table "quiz_choices", force: :cascade do |t|
  t.integer "display_order", default: 0, null: false
  t.check_constraint "`display_order` between 1 and 4", name: "display_order_check"
end
```

ref: https://kazuya-engineer.com/2024/02/03/rails-migrations-allow-specific-values/

If it is to be supported, it would be in the form of adding it to `extractTableDetails` function.💭
I don't think we must necessarily address this in this pull request!

Reply:
Thanks, I missed the method.
Yes, it seems necessary to support, but maybe in a separate PR. I'll create an issue later.

Reply:
Thank you so much! ✨